tests/tests_pytorch/strategies/test_xla_fsdp.py

```diff
@@ -119,16 +136,39 @@ def test_rank_properties_access(xla_available):
 )
 def test_xla_fsdp_strategy_checkpoint(tmpdir, state_dict_type, ckpt_location):
     """Test to ensure that checkpoint is saved correctly when using TPUs, and all stages can be run."""
-    model = TestFSDPModel()
+    from torch_xla.distributed.fsdp.wrap import always_wrap_policy
+
+    model = TestFSDPModel(use_auto_wrap_policy=True)
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        accelerator="tpu",
+        strategy=XLAFSDPStrategy(auto_wrap_policy=always_wrap_policy, state_dict_type=state_dict_type),
+        precision="bf16-true",
+        max_epochs=1,
+    )
+    _run_multiple_stages(trainer, model, state_dict_type, True, os.path.join(tmpdir, ckpt_location))
 
+
+@RunIf(min_torch="2.0", tpu=True, standalone=True)
+@pytest.mark.parametrize(
+    ("state_dict_type", "ckpt_location"),
+    [
+        ("full", "lightning_logs/version_0/checkpoints"),
+        ("sharded", "lightning_logs/version_0/checkpoints/epoch=0-step=64.ckpt"),
+    ],
+)
+def test_xla_fsdp_manual_wrap(tmpdir, state_dict_type, ckpt_location):
+    """Test to ensure that all stages can be run when manually wrapping the model."""
+    model = TestFSDPModel(use_auto_wrap_policy=False)
     trainer = Trainer(
         default_root_dir=tmpdir,
         accelerator="tpu",
         strategy=XLAFSDPStrategy(state_dict_type=state_dict_type),
         precision="bf16-true",
         max_epochs=1,
     )
-    _run_multiple_stages(trainer, model, state_dict_type, os.path.join(tmpdir, ckpt_location))
+
+    _run_multiple_stages(trainer, model, state_dict_type, False, os.path.join(tmpdir, ckpt_location))
```

Comment:
Couldn't this be?

```suggestion
    _run_multiple_stages(trainer, model, state_dict_type, state_dict_type == "sharded", os.path.join(tmpdir, ckpt_location))
```

Reply:
Since there is another test that checks checkpoint and those tests take longer, I wanted to test manual wrap without checkpointing. IMO testing one model wrapping configuration is good enough for testing checkpointing but if we need more converage we can make this change.